app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/SyncAccountsWithConnectorExtensionUseCase.kt

```diff
@@ -0,0 +1,114 @@
+@file:Suppress("LongParameterList")
+
+package com.babylon.wallet.android.domain.usecases.p2plink
+
+import com.babylon.wallet.android.data.dapp.PeerdroidClient
+import com.babylon.wallet.android.data.dapp.model.Account
+import com.babylon.wallet.android.data.dapp.model.ConnectorExtensionExchangeInteraction
+import com.babylon.wallet.android.data.repository.p2plink.P2PLinksRepository
+import com.radixdlt.sargon.Hash
+import com.radixdlt.sargon.RadixConnectPurpose
+import com.radixdlt.sargon.extensions.bytes
+import com.radixdlt.sargon.extensions.hash
+import com.radixdlt.sargon.extensions.hex
+import com.radixdlt.sargon.extensions.init
+import com.radixdlt.sargon.extensions.string
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import rdx.works.core.hash
+import rdx.works.core.preferences.PreferencesManager
+import rdx.works.core.sargon.activeAccountsOnCurrentNetwork
+import rdx.works.peerdroid.data.PeerdroidConnector
+import rdx.works.peerdroid.domain.PeerConnectionStatus
+import rdx.works.profile.domain.GetProfileUseCase
+import timber.log.Timber
+import javax.inject.Inject
+
+class SyncAccountsWithConnectorExtensionUseCase @Inject constructor(
+    private val p2pLinksRepository: P2PLinksRepository,
+    private val getProfileUseCase: GetProfileUseCase,
+    private val peerdroidConnector: PeerdroidConnector,
+    private val peerdroidClient: PeerdroidClient,
+    private val preferencesManager: PreferencesManager,
+    private val json: Json
+) {
+
+    suspend operator fun invoke() {
+        val connectionIdsFlow = peerdroidConnector.peerConnectionStatus
+            .filter { connectionStatuses ->
+                connectionStatuses.isNotEmpty() &&
+                    !connectionStatuses.any { it.value == PeerConnectionStatus.CONNECTING }
+            }
```

Comment:
the only thing that I would change is to expose these functions from `PeerdroidClient` since this client has a reference to `PeerdroidConnector`. And then no need to pass the `PeerdroidConnector` here.

That's the purpose of the `PeerdroidClient` anyway!

Reply:
Sounds good!